extensions/sockets/src/test/java/org/mule/extension/socket/SendEncodedStringTestCase.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.socket;
+
+import static org.junit.Assert.assertEquals;
+import org.mule.runtime.api.message.MuleMessage;
+import org.mule.runtime.core.util.IOUtils;
+
+import java.io.InputStream;
+
+import org.junit.Test;
+
+public class SendEncodedStringTestCase extends SocketExtensionTestCase
+{
+    private static final String WEIRD_CHAR_MESSAGE = "This is a messag\u00ea with weird chars \u00f1.";
+    private static final String ENCODING = "iso-8859-1";
```

Comment:
this test is flaky. You're assuming that the default encoding is not iso-8859 just because in your machine this isn't the case. What you should do is to make this dynamic and make sure that you pick a supported Charset which is not your machine's default. 

Look at the FileWriteTestCase for inspiration